tests/utils/test_env.py

```diff
@@ -619,6 +621,55 @@ def err_on_rm_venv_only(path, *args, **kwargs):
     m.side_effect = original_rmtree  # Avoid teardown using `err_on_rm_venv_only`
 
 
+def test_remove_keeps_dir_if_not_deleteable_os_err_20(
```

Comment:
Let's parameterize this whole test for different `errno` constants instead of doing this copy-and-paste.